packages/api/src/command/medical/admin/hie-overview.ts

```diff
@@ -0,0 +1,240 @@
+  response_data: OutboundPatientDiscoveryResp | undefined;
+  name: string | undefined | null;
+};
+
+export async function getHieOverview(
+  patientId: string,
+  debugLevel: DebugLevel
+): Promise<HiePatientOverview | undefined> {
+  const queryResp: DbQueryResponseRecord[] = await getDataFromDb(patientId, debugLevel);
+  const [respErrors, respSuccess] = partition(
+    queryResp,
+    r => r.response_data?.patientMatch === false
+  );
+  const patientData = respSuccess[0]?.patient_data;
+
+  const errors: CqPdErrorConsolidated[] = respErrors.reduce(
+    (acc: CqPdErrorConsolidated[], row: DbQueryResponseRecord) => {
+      const rowErrorDetail = (
+        row.response_data?.operationOutcome?.issue[0]?.details as { text: string | undefined }
+      )?.text;
+      if (!rowErrorDetail) return acc;
+      const gatewayName = getGatewayName(row);
+      const existingErrorEntry = acc.find(a => a.error === rowErrorDetail);
+      if (existingErrorEntry) {
+        existingErrorEntry.gatewayNames.push(gatewayName);
+        return acc;
+      }
+      acc.push({
+        error: rowErrorDetail,
+        gatewayNames: [gatewayName],
+      });
+      return acc;
+    },
+    new Array<CqPdErrorConsolidated>()
+  );
+
+  const hieOverview = respSuccess.reduce(
+    (acc: HiePatientOverview | undefined, row: DbQueryResponseRecord) => {
+      if (!row.response_data) return acc;
+      const patientOfSuccessfulResp = responseToPatient(row.response_data, patientData);
+      if (!patientOfSuccessfulResp) {
+        console.log(`Skipping response with missing patient data: ${row.response_data.id}`);
+        return acc;
+      }
+      const gatewayName = getGatewayName(row);
+      const gateway = responseToGateway(row.response_data, gatewayName);
+      const success = {
+        patient: patientOfSuccessfulResp,
+        gateway,
+      };
+      if (acc) {
+        acc.carequality.success.push(success);
+        return acc;
+      }
+      const cqLinks = responseToCqLinks(row);
+      return {
+        cxId: row.cx_id,
+        patientId: row.id,
+        patientData,
+        carequality: {
+          links: cqLinks,
+          success: [success],
+          errors,
+        },
+      };
+    },
+    undefined
+  );
+  return hieOverview;
+}
+
+async function getDataFromDb(
+  patientId: string,
+  debugLevel: DebugLevel
+): Promise<DbQueryResponseRecord[]> {
+  const matchQuery = debugLevel === "success" ? `and r.data->>'patientMatch' = 'true'` : "";
+  const query =
+    debugLevel !== "info"
+      ? `
+          select p.cx_id, p.id, p.data as patient_data, pd.data as cq_links, r.data as response_data, d.name as gateway_name
+          from patient p
+            left outer join cq_patient_data pd on pd.id = p.id
+            left join patient_discovery_result r on r.patient_id::text = p.id ${matchQuery}
+            left join cq_directory_entry d on d.id = r.data->>'gatewayHomeCommunityId'
+          where p.id = ':patientId'
```

Comment:
I think wrapping parameter in string either makes the parametrization not work or vulnerable to injections again: https://sequelize.org/docs/v7/querying/raw-queries/#%EF%B8%8F-dont-put-parameters-in-strings

```where p.id = :patientId```

is the recommended syntax without the quoting syntax it seems. Ditto below

Reply:
Good catch! This seem specific to v7, though. [v6 doesn't mention this](https://sequelize.org/docs/v6/core-concepts/raw-queries/#replacements) and I tested local and couldn't inject SQL, with or without quotes.

But considering the potential risk, I agree we should always use replacement without quotes - added to [code standards](https://www.notion.so/metriport/Development-Best-Practices-bb3574d3c83d46dc9914a9e5c5542ca7?pvs=4#83b0008360724a50b805d659d77dc9e5).

This is a big issue with Sequelize IMO. It either guarantees safety against SQL injection or not... relying on whether the named parameter is within quotes or not is brittle and error prone.

Reply:
Awesome. Yah I agree thats kinda brittle. Great adding to code standards and just something to always keep an eye out for whenever PRs have raw sql in them